src/MeoAssistant/Utils/AsstUtils.hpp

```diff
@@ -25,6 +25,29 @@ namespace asst::utils
     template <typename char_t = char>
     using pair_of_string_view = std::pair<std::basic_string_view<char_t>, std::basic_string_view<char_t>>;
 
+    template <std::contiguous_iterator It, std::sized_sentinel_for<It> End,
+              typename Ret = std::basic_string_view<std::remove_cvref_t<std::iter_value_t<It>>>>
+    requires(!std::convertible_to<End, typename Ret::size_type>)
+    inline Ret make_string_view(It beg, End end)
+    {
+        return { std::to_address(beg), static_cast<typename Ret::size_type>(end - beg) };
+    }
+
+    template <std::contiguous_iterator It, typename Sz,
+              typename Ret = std::basic_string_view<std::remove_cvref_t<std::iter_value_t<It>>>>
+    requires(std::convertible_to<Sz, typename Ret::size_type>)
+    inline Ret make_string_view(It beg, Sz cnt)
+    {
+        return { std::to_address(beg), static_cast<typename Ret::size_type>(cnt) };
+    }
+    
+    template <ranges::range Rng>
+    requires(requires(Rng rng) { make_string_view(rng.begin(), rng.end()); })
```

Comment:
```suggestion
    requires(requires(Rng rng) { make_string_view(&*rng.begin(), ranges::distance(rng)); })
    inline auto make_string_view(Rng rng)
    {
        return make_string_view(&*rng.begin(), ranges::distance(rng));
    }
```

我在[这里](https://stackoverflow.com/a/48403210)搜到了一个解决方案，可以直接从 ranges 转换成 string_view。试了下 clang + range-v3 和 gcc 都能编译过

@horror-proton

Reply:
确实，但不安全，还是得用宏来控制一下

Reply:
我的建议是，在启用 ASST_USE_RANGES_RANGE_V3 的时候，把 contiguous_iterator 直接换成 forward_iterator（

Reply:
> 我的建议是，在启用 ASST_USE_RANGES_RANGE_V3 的时候，把 contiguous_iterator 直接换成 forward_iterator（

我试过了, 那个迭代器似乎不能 `to_address` 只能 `operator*`

Reply:
草，看到晚了，推上去了

Reply:
那能减法么，`operator-`

Reply:
减法肯定不行